sdk-extension/opentelemetry-sdk-extension-aws/src/opentelemetry/sdk/extension/aws/trace/aws_xray_id_generator.py

```diff
@@ -56,29 +56,25 @@
 .. _trace ID format: https://docs.aws.amazon.com/xray/latest/devguide/xray-api-sendingdata.html#xray-api-traceids
 """
 
-import random
 import time
+import random
 
-from opentelemetry.sdk.trace.id_generator import IdGenerator, RandomIdGenerator
-
-
-class AwsXRayIdGenerator(IdGenerator):
-    """Generates tracing IDs compatible with the AWS X-Ray tracing service. In
-    the X-Ray system, the first 32 bits of the `TraceId` are the Unix epoch time
-    in seconds. Since spans (AWS calls them segments) with an embedded timestamp
-    more than 30 days ago are rejected, a purely random `TraceId` risks being
-    rejected by the service.
 
+class AwsXRayIdGenerator():
+    """Generates tracing IDs compatible with the AWS X-Ray tracing service.
     See: https://docs.aws.amazon.com/xray/latest/devguide/xray-api-sendingdata.html#xray-api-traceids
+    See Same implementation in Javascript: https://github.com/aws-observability/aws-otel-js/tree/main/packages/opentelemetry-id-generator-aws-xray
     """
-
-    random_id_generator = RandomIdGenerator()
-
-    def generate_span_id(self) -> int:
-        return self.random_id_generator.generate_span_id()
-
-    @staticmethod
-    def generate_trace_id() -> int:
-        trace_time = int(time.time())
-        trace_identifier = random.getrandbits(96)
-        return (trace_time << 96) + trace_identifier
+    TIME = time
+    TRACE_ID_LENGTH = 32
+    SPAN_ID_LENGTH = 16
+
+    # Ex: 0c6d1c759808e783
```

Comment:
Hi @schanjr - the interface for ID generation in python and JS are quite different, in python the return values must be integers

https://github.com/open-telemetry/opentelemetry-python/blob/main/opentelemetry-sdk/src/opentelemetry/sdk/trace/id_generator.py#L21

So I think this new implementation won't actually work when the SDK tries to use it and gets a string back.

I don't see anything wrong with the previous implementation, can you describe where you're seeing wrong IDs?

Reply:
@anuraaga Thank you for taking a look at this. Upon inspecting the logs again, I realized the final output did indeed convert the int values to hex at some point. The OTEL collector in turn converts the hex to xray format before pushing to aws. I was a bit surprised as the JS version of SDK uses the same OTEL collector image and pushes the actual xray format to the collector. The collector did not need to do the conversion. 


 Sorry for the trouble :) 


span console output
```json
{
    "name": "/health asgi",
    "context": {
        "trace_id": "0x606a8c10b37a1e9cad7c550289dae5ae",
        "span_id": "0xe6f70a22665060ae",
        "trace_state": "[]"
    },

```

otel collector image 
```
    "{\"name\":\"fastapi-app\",\"id\":\"e6f70a22665060ae\",\"start_time\":1617595408.623005,\"trace_id\":\"1-606a8c10-b37a1e9cad7c550289dae5ae\",\"end_time\":1617595408.623856,\"http\":{\"request\":{\"method\":\"GET\",\"url\":\"http://127.0.0.1:31000/health\",\"user_agent\":\"PostmanRuntime/7.26.10\",\"client_ip\":\"127.0.0.1\"},\"response\":{\"status\":200,\"content_length\":0}},\"fault\":false,\"error\":false,\"aws\":{\"xray\":{\"sdk\":\"opentelemetry for python\",\"sdk_version\":\"1.0.0\",\"auto_instrumentation\":false}},\"metadata\":{\"default\":{\"http.flavor\":\"1.1\",\"http.route\":\"/health\",\"net.host.port\":\"\",\"otel.resource.service.name\":\"fastapi-app\",\"otel.resource.telemetry.sdk.language\":\"python\",\"otel.resource.telemetry.sdk.name\":\"opentelemetry\",\"otel.resource.telemetry.sdk.version\":\"1.0.0\"}}}\n"
  ]

```